src/librustdoc/html/layout.rs

```diff
@@ -34,17 +35,23 @@ pub(crate) struct Page<'a> {
 }
 
 impl<'a> Page<'a> {
-    pub(crate) fn get_static_root_path(&self) -> &str {
-        self.static_root_path.unwrap_or(self.root_path)
+    pub(crate) fn get_static_root_path(&self) -> String {
+        match self.static_root_path {
```

Comment:
I just thought about this: instead of recomputing the path every time, wouldn't it be better to instead compute this information even before creating the `Page` struct?